### PR TITLE
feat(octavia): add link to pool from listeners list

### PR DESCRIPTION
**Pull request by JacquesLarique**
| Question         | Answer
| ---------------- | ---
| Branch?          |  `feat/octavia-batch-1` 
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-12602
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] ~~Only FR translations have been updated~~ (n/a)
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~ (n/a)

## Description

Added a link to the pool detail from the listeners list and adjusted the links in the listeners list

### CODE REVIEW

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/listeners/list/template.html [hunk 1]
```diff
@@ -20,7 +20,12 @@
...
         data-prevent-customization
     >
-        <a data-ng-click="$ctrl.goToListenerEdition($row)">
+        <a
+            data-ng-href="{{:: $ctrl.getListenerEditionLink($row)}}"
+            data-track-on="click"
+            data-track-name="{{:: $ctrl.listenerEditionTrackingHit}}"
+            data-track-type="action"
+        >
             {{ ::$value }}
```
Comment (darsene) [inline packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/listeners/list/template.html:29]:
```suggestion
            <span data-ng-bind="::$value"></span>
```

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/listeners/list/template.html [hunk 2]
```diff
@@ -32,7 +37,16 @@
...
         data-helper=":: 'octavia_load_balancer_listeners_default_pool_helper' | translate"
+        data-prevent-customization
     >
+        <a
+            data-ng-href="{{:: $ctrl.getPoolDetailLink($row)}}"
+            data-track-on="click"
+            data-track-name="{{:: $ctrl.poolDetailTrackingHit}}"
+            data-track-type="action"
+        >
+            {{ ::$value }}
```
Comment (darsene) [inline packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/listeners/list/template.html:48]:
```suggestion
            <span data-ng-bind="::$value"></span>
```

**Review by darsene — APPROVED** [review]

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/listeners/edit/index.js [hunk 3]
```diff
@@ -10,7 +10,7 @@ angular.module(moduleName, ['ui.router', 'oc.lazyLoad']).config(
     $stateProvider.state(
       'octavia-load-balancer.loadbalancer.listeners.edit.**',
       {
-        url: '/edit',
+        url: '/edit?:listenerId',
```
Comment (anooparveti) [inline packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/listeners/edit/index.js:13]:
```suggestion
        url: '/edit/:listenerId',
```

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/listeners/edit/routing.js [hunk 4]
```diff
@@ -4,7 +4,7 @@ import { TRACKING_NAME } from '../../constants';
 
 export default /* @ngInject */ ($stateProvider) => {
   $stateProvider.state('octavia-load-balancer.loadbalancer.listeners.edit', {
-    url: '/edit?listenerId',
+    url: '/edit?:listenerId',
```
Comment (anooparveti) [inline packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/listeners/edit/routing.js:7]:
```suggestion
    url: '/:listenerId/edit',
```

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/listeners/list/component.js [hunk 5]
```diff
@@ -5,8 +5,12 @@ export default {
   bindings: {
     listeners: '<',
     goToListenerCreation: '<',
+    getListenerEditionLink: '<',
     goToListenerEdition: '<',
     goToListenerDeletion: '<',
+    getPoolDetailLink: '<',
+    listenerEditionTrackingHit: '<',
```
Comment (anooparveti) [inline packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/listeners/list/component.js:12]: **nitpick(seperation-of-concerns):** take `trackBase` as input to the component and define `edit` related base in the edit component itself

**Review by anooparveti — APPROVED** [review]

**Review by anooparveti — APPROVED** [review]

**Pull request merged**